wrapanapi/systems/ec2.py

```diff
@@ -804,17 +807,24 @@ def object_exists_in_bucket(self, bucket_name, object_key):
         objects = [o for o in bucket.objects.all() if o.key == object_key]
         return any(objects)
 
-    def delete_s3_bucket(self, bucket_name):
-        """TODO: Force delete - delete all objects and then bucket"""
-        bucket = self.s3_connection.Bucket(bucket_name)
-        self.logger.info("Trying to delete bucket '%s'", bucket_name)
-        try:
-            bucket.delete()
-            self.logger.info("Success: bucket '%s' was deleted.", bucket_name)
-            return True
-        except Exception:
-            self.logger.exception("Bucket '%s' deletion failed", bucket_name)
-            return False
+    def delete_s3_buckets(self, bucket_names):
+        """ Deletes specified bucket(s) with keys """
+        if isinstance(bucket_names, (set, list, tuple)):
+            buckets = [self.s3_connection.Bucket(obj_name) for obj_name in bucket_names]
+        else:
+            raise TypeError("Object is not iterable.")
+        for bucket in buckets:
+            self.logger.info("Trying to delete bucket '%s'", bucket)
+            keys = [obj.key for obj in bucket.objects.all()]
+            if keys:
+                self.delete_objects_from_s3_bucket(bucket.name, keys)
```

Comment:
QUESTION: since you handle exceptions during bucket deletion. Shouldn't your try/except be expanded to this delete call as well ?

Reply:
Moved if block under try-exception